### PR TITLE
Fix #3712

### LINKED ISSUE #3712: Selenium exception: Cannot read property 'edges' of undefined
Hi everyone, i just git pulled then I got this error while using **interact_user_followers**

```
Traceback (most recent call last):
  File "like.py", line 57, in <module>
    session.interact_user_followers(['user'], amount=1500, randomize=False)
  File "/Users/users/Desktop/instagram/InstaPy/instapy/instapy.py", line 2799, in interact_user_followers
    self.user_interact_media)
  File "/Users/users/Desktop/instagram/InstaPy/instapy/instapy.py", line 2189, in interact_by_users
    self.logger))
  File "/Users/users/Desktop/instagram/InstaPy/instapy/like_util.py", line 537, in check_link
    ''', user_name)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 636, in execute_script
    'args': converted_args})['value']
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: Cannot read property 'edges' of undefined
  (Session info: chrome=)
  (Driver info: chromedriver= (),platform=Mac OS X 10.13.3 x86_64)
```

This is the code I am using:
```session.set_simulation(enabled=False)
session.set_skip_users(skip_private=True,
                       private_percentage=100,
                       skip_no_profile_pic=False,
                       no_profile_pic_percentage=100,
                       skip_business=False,
                       business_percentage=100,
                       skip_business_categories=[],
                       dont_skip_business_categories=[])
session.set_user_interact(amount=2, randomize=True, percentage=100, media='Photo')
session.set_do_like(enabled=True, percentage=100)
session.interact_user_followers(['user'], amount=1500, randomize=False)
```

### PR DESCRIPTION
**Pull request by ReagentX**

### CODE REVIEW

**Comment by sionking** [comment]
what about "edge_media_to_comment" you ignore it for some users that still in A test

File: instapy/like_util.py [hunk 1]
```diff
@@ -568,7 +581,7 @@ def check_link(browser, post_link, dont_like, mandatory_words,
     """If the image still has no description gets the first comment"""
     if image_text is None:
         if graphql:
-            image_text = media['edge_media_to_comment']['edges']
```
Comment (sionking) [inline instapy/like_util.py:571]: you remove old style here

Comment (sionking) [inline instapy/like_util.py:571]: you remove old style here

Comment (sionking) [inline instapy/like_util.py:571]: you remove old style here

Comment (sionking) [inline instapy/like_util.py:571]: you remove old style here

Comment (sionking) [inline instapy/like_util.py:571]: you remove old style here

**Comment by ReagentX** [comment]
@sionking you are looking at the wrong branch, check my pr.

**Comment by timgrossmann** [comment]
@sionking Good to see that your still up and healthy 😊 
Are you just not active on Discord or are you reading everything in silence? 😉

**Review by w4ngel — APPROVED** [review]
Looks good from me!

**Review by sionking — COMMENTED** [review]
> @sionking Good to see that your still up and healthy 😊
> Are you just not active on Discord or are you reading everything in silence? 😉

I have too busy, and as long as the instagram page did not change I keep silence :)

**Comment by timgrossmann** [comment]
@sionking The silent knight! 😉 

@w4ngel I'm not sure about the thing @sionking pointed out. We are not adding the possibility that there are still users with the other view. 
We just replace it with the new one

**Comment by dsmh** [comment]
Is this patch going to be merged at some point? I have several bots down cause this bug, however the manual patch is not working for me. It would be great to have the patch include in master branch to pull it. Thanks in advance.

**Comment by w4ngel** [comment]
> @w4ngel I'm not sure about the thing @sionking pointed out. We are not adding the possibility that there are still users with the other view.

It seems to me that this accounts for both? I could be overlooking something but this just tries a different method if it doesn't work?

**Comment by ReagentX** [comment]
This accounts for both and is forward looking as future changes can just be added to the `options` list. Don’t know why people are saying this doesn’t account for both strings.

**Comment by timgrossmann** [comment]
@dsmh Sorry for the delay.

@w4ngel Looking at the discussion with @sionking in the mentioned issue, image text, for some users is
```
image_text = media['edge_media_to_comment']['edges']
```
```
image_text = media[media_edge_string]['edges']
```

In this PR we are just using the new (lower one) and replace the old (upper one) which might still be used (Works fine with my account with the old version)
So it will break the logic for users that are not part of the A/B test.

We could introduce both by using try/catch for now. Making sure that once the "new" approach fails, we still try the "old" approach 

Sorry for the confusion, hope I could explain it more clear now

Edit: @ReagentX I just saw your comment. I'm still not sure since I can't see the "old" GraphQL object path in your solution

**Comment by ReagentX** [comment]
Line 470 @timgrossman

**Comment by ReagentX** [comment]
`media_edge_string` is resolved to whatever string is present: non A/B accounts get the old string, A/B accounts get the new string. That’s why I added `get_media_edge_comment_string()`. For future changes, the string can just get added to that list. Possibly this should be abstracted more in the future and most of the hardcoded strings should be resolved like this.

**Comment by ReagentX** [comment]
I see the problem now, I misread the scope. Should be good to go now, my bad.

**Comment by timgrossmann** [comment]
@ReagentX There we go, sorry for the confusion.
Nice way of dealing with that! :+1:

**Comment by timgrossmann** [comment]
@sionking Would you second this PR now?

**Comment by sionking** [comment]
I am no JS expert but it seems ok!

**Comment by timgrossmann** [comment]
@sionking Thank you for the review.

@ReagentX Thank you very much for this fix.

**Pull request merged**